internal/gcs/container.go

```diff
@@ -19,15 +22,34 @@ const hrComputeSystemDoesNotExist = 0xc037010e
 // Container implements the cow.Container interface for containers
 // created via GuestConnection.
 type Container struct {
-	gc        *GuestConnection
-	id        string
-	notifyCh  chan struct{}
-	closeCh   chan struct{}
+	gc *GuestConnection
+	id string
+	// Blocks Wait() until waitBackground() returns or Close() is called.
+	// Does not represent the status of the container in the guest
+	waitBlock chan struct{}
+	// set only in closeOnce, read after <-waitBlock returns
+	waitError error
 	closeOnce sync.Once
+	// The channel notifications are received on from the guest connection
+	notifyCh chan notifications.Message
```

Comment:
why aren't we doing something like `notifyChans` here for notifyCh and closeCh?

Reply:
I didnt think too, I'll update accordingly